server/mergin/auth/bearer.py

```diff
@@ -3,17 +3,28 @@
 # SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-MerginMaps-Commercial
 
 import hashlib
+from datetime import datetime, timezone
 from itsdangerous import URLSafeTimedSerializer
+from itsdangerous.exc import SignatureExpired, BadSignature
 from flask.sessions import TaggedJSONSerializer
 
 
-def decode_token(secret_key, salt, token, max_age=None):
+def decode_token(secret_key, salt, token):
     serializer = TaggedJSONSerializer()
     signer_kwargs = {"key_derivation": "hmac", "digest_method": hashlib.sha1}
     s = URLSafeTimedSerializer(
         secret_key, salt=salt, serializer=serializer, signer_kwargs=signer_kwargs
     )
-    return s.loads(token, max_age=max_age)
+    token_data = s.loads(token)
+    try:
+        expire = datetime.fromisoformat(token_data.get("expire"))
+    except (ValueError, TypeError):
+        raise BadSignature("Invalid token")
+
+    if expire < datetime.now(timezone.utc):
```

Comment:
When we build the token, we use `pytz.utc`, here when decoding we use `timezone.utc` -- I guess that is allright, right @varmar05?